Models/Report.cs

```diff
@@ -10,62 +10,37 @@ public class Report {
 		public Report() {
 			List<Consumable> vanillaConsumables = EnumUtil
 				.AllCases<ConsumableItem>()
-				.Select(i => new VanillaConsumable(i))
-				.Select(vc => new Consumable(vc))
+				.Select(ci => new VanillaConsumable(ci))
+				.Select(vc => new Consumable(vanillaItemId: vc.ItemId, type: vc.Type, currentCount: vc.CurrentCount, totalCount: vc.TotalCount))
 				.ToList();
 
 			ConsumablesList.Add(new(VanillaConsumableMod, vanillaConsumables));
-
-			//public class TestConsumableMod : IConsumableMod {
-			//	public string ModTabName => "Test Mod";
-
-			//	public string ModTabTexturePath() {
-			//		return "Terraria/Images/Item_4760";
-			//	}
-
-			//	public TestConsumableMod() { }
-			//}
-
-			//List<IConsumable> fakeList = [
-			//	new VanillaConsumable(ConsumableItem.minecartUpgradeKit),
-			//	new VanillaConsumable(ConsumableItem.peddlersSatchel),
-			//	new VanillaConsumable(ConsumableItem.artisanLoaf)
-			//];
-			//ConsumablesList.Add(new(new TestConsumableMod(), fakeList));
 		}
 
-		public static bool AddModtoList(ConsumableMod mod) {
-			foreach (ConsumablesEntry entry in ConsumablesList) {
-				if (entry.Mod.ModTabName == mod.ModTabName) {
-					return false;
-				}
+		internal static ConsumablesEntry GetModEntryOrAddIfNeeded(ConsumableMod mod) {
+			ConsumablesEntry foundEntry = ConsumablesList.Find(e => e.Mod.ModTabName == mod.ModTabName);
+			if (foundEntry != null) {
+				return foundEntry;
+			} else {
+				ConsumablesEntry newEntry = new(mod, []);
+				ConsumablesList.Add(newEntry);
+				return newEntry;
 			}
-
-			// at this point, we don't have the mod registered yet
-			// Add new entry with empty consumables arr
-			ConsumablesList.Add(new(mod, []));
-			return true;
 		}
 
-		public static bool AddConsumableToList(string modName, Consumable consumable) {
-			foreach (ConsumablesEntry entry in ConsumablesList) {
-				if (entry.Mod.ModTabName == modName) {
-					// mod exists
-					foreach (Consumable c in entry.Consumables) {
-						if (c.DisplayText == consumable.DisplayText) {
-							// consumable already exists, exit
-							return false;
-						}
-					}
+		public static bool AddConsumableToList(ConsumableMod mod, Consumable consumable) {
+			ConsumablesEntry entry = GetModEntryOrAddIfNeeded(mod);
 
-					// consumable does not exist, add it
-					entry.Consumables.Add(consumable);
-					return true;
+			foreach (Consumable c in entry.Consumables) {
+				if (c.DisplayText == consumable.DisplayText) {
+					// consumable already exists, exit
```

Comment:
Probably better to store a unique identifier (such as the item's type (ModItem.Type / id) to then check for equality